diff_cover/diff_quality_tool.py

```diff
@@ -222,20 +224,33 @@ def main(argv=None, directory=None):
         last_char = user_options[-1]
         if first_char == last_char and first_char in ('"', "'"):
             user_options = user_options[1:-1]
+    reporter = None
     driver = QUALITY_DRIVERS.get(tool)
-
-    if driver is not None:
-        # If we've been given pre-generated reports,
-        # try to open the files
+    if driver is None:
+        # The requested tool is not built into diff_cover. See if another Python
+        # package provides it.
+        pm = pluggy.PluginManager('diff_cover')
+        pm.add_hookspecs(hookspecs)
+        pm.load_setuptools_entrypoints('diff_cover')
+        for name, reporter_fn in pm.hook.diff_cover_report_quality():
+            if name == tool:
+                reporter = reporter_fn()
+                break
+
+    if reporter is not None or driver is not None:
         input_reports = []
-
-        for path in arg_dict['input_reports']:
-            try:
-                input_reports.append(open(path, 'rb'))
-            except IOError:
-                LOGGER.warning("Could not load '{}'".format(path))
         try:
-            reporter = QualityReporter(driver, input_reports, user_options)
+            if driver is not None:
```

Comment:
These changes were a bit more complex than I hoped. It seems to me the current design assumes that a `reporter` has an underlying `driver` object that runs an external command. Plugins are a "tighter" integration -- rather than an external command, we have `diff-quality` *directly* calling the Python code in the other tool. It might be good someday to clean this up a bit, but in this PR I was trying to make the minimum changes to support both the old (external command) and new (plugin) ways of integrating tools.